src/vg_active_segment.erl

```diff
@@ -68,17 +69,34 @@ write(Topic, Partition, RecordSet) ->
             R -> R
         end
     catch _:{noproc, _} ->
-            create_retry(Topic, Partition, RecordSet);
+            create_retry(Topic, Partition, head, RecordSet);
           error:badarg ->  %% is this too broad?  how to restrict?
-            create_retry(Topic, Partition, RecordSet);
+            create_retry(Topic, Partition, head, RecordSet);
           exit:{timeout, _} ->
             {error, timeout}
     end.
 
-create_retry(Topic, Partition, RecordSet)->
+write(Topic, Partition, ExpectedId, RecordSet) ->
```

Comment:
can you rewrite write/3 to just call write/4?

Reply:
Done and rebased.